### PR TITLE
BaSP-TG-9-B: Admin - Add and Edit

### PR DESCRIPTION
**Pull request by FabianDutraSilva**

### CODE REVIEW

File: src/Components/Admins/index.js [hunk 1]
```diff
@@ -1,9 +1,33 @@
...
+      method: 'DELETE'
+    });
+    saveAdmins([...admins.filter((newListItem) => newListItem._id !== id)]);
+  };
+
   return (
     <section className={styles.container}>
       <h2>Admins</h2>
+      <Table admins={admins} deleteAdmin={deleteAdmin} />
+      <a href={`admins/form`}>
```
Comment (Maxi-SR) [inline src/Components/Admins/index.js:28]: Try ` href=" admins/form" `

File: src/Components/Admins/modal-delete.module.css [hunk 2]
```diff
@@ -0,0 +1,36 @@
+.container {
+    position: fixed;
+    left: 0;
+    top: 0;
+    right: 0;
+    bottom: 0;
+    justify-content: center;
+    align-items: center;
+    background-color: rgba(0,0,0,0.5);
```
Comment (Maxi-SR) [inline src/Components/Admins/modal-delete.module.css:9]: they recommend us to use colors in hexadecimal format

File: src/Components/Admins/modalDelete.jsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import styles from './modal-delete.module.css';
```
Comment (Maxi-SR) [inline src/Components/Admins/modalDelete.jsx:2]: I think that is better use filenames in camel case

File: src/Components/Admins/modalDelete.jsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+const Modal = (props) => {
```
Comment (Maxi-SR) [inline src/Components/Admins/modalDelete.jsx:4]: Maybe you can use destructuring here

File: src/Components/Admins/table.module.css [hunk 5]
```diff
@@ -0,0 +1,18 @@
+.table {
+    margin: auto;
+    box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);
+    border-collapse:collapse;
+  }
+
+  .table thead{
+  background-color: rgba(55, 56, 103, 0.1);;
```
Comment (Maxi-SR) [inline src/Components/Admins/table.module.css:8]: Use hex color here

File: src/Components/Admins/form.jsx [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+    name: '',
+    lastName: '',
+    email: '',
+    password: '',
+    status: ''
+  });
+
+  useEffect(async () => {
+    const params = new URLSearchParams(window.location.search);
+    const adminId = params.get('id');
```
Comment (DamianPalavecino) [inline src/Components/Admins/form.jsx:15]: Here you can set the constants 'params' and 'taskId' as global variables so you don't have to repeat them at line 42

File: src/Components/Admins/modalDelete.jsx (hunk 3 above)
Comment (FabianDutraSilva) [reply src/Components/Admins/modalDelete.jsx:2]: I think css file names should be in lower case. Ex: modal-delete

File: src/Components/Admins/index.js (hunk 1 above)
Comment (FabianDutraSilva) [reply src/Components/Admins/index.js:28]: solved

File: src/Components/Admins/modal-delete.module.css (hunk 2 above)
Comment (FabianDutraSilva) [reply src/Components/Admins/modal-delete.module.css:9]: we can keep it if it's due to opacity, so i'm leaving it like that

File: src/Components/Admins/modalDelete.jsx (hunk 4 above)
Comment (FabianDutraSilva) [reply src/Components/Admins/modalDelete.jsx:4]: done

File: src/Components/Admins/table.module.css (hunk 5 above)
Comment (FabianDutraSilva) [reply src/Components/Admins/table.module.css]: we are keeping rgba beacuse of opacity use

Comment (FabianDutraSilva) [reply src/Components/Admins/table.module.css]: extra ;; fixed

File: src/Components/Admins/form.jsx (hunk 6 above)
Comment (FabianDutraSilva) [reply src/Components/Admins/form.jsx]: done

File: package.json [hunk 7]
```diff
@@ -6,6 +6,7 @@
     "@testing-library/jest-dom": "^5.15.1",
     "@testing-library/react": "^11.2.7",
     "@testing-library/user-event": "^12.8.3",
+    "cors": "^2.8.5",
```
Comment (martumoser) [inline package.json:9]: Not necessary to install cors on the frontend

File: src/Components/Admins/table.module.css [hunk 8]
```diff
@@ -0,0 +1,18 @@
+.table {
+    margin: auto;
+    box-shadow: 0 0 20px rgba(0, 0, 0, 0.15);
+    border-collapse:collapse;
+  }
```
Comment (martumoser) [inline src/Components/Admins/table.module.css:5]: Please fix indentation on this file

**Review by martumoser — CHANGES_REQUESTED** [review]

File: package.json (hunk 7 above)
Comment (FabianDutraSilva) [reply package.json:9]: Same mistake, sry. Fixed

File: src/Components/Admins/table.module.css (hunk 8 above)
Comment (FabianDutraSilva) [reply src/Components/Admins/table.module.css:5]: Done!

File: src/Components/Admins/form.jsx [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+  const [inputValue, setInputValue] = useState({
+    name: '',
+    lastName: '',
+    email: '',
+    password: '',
+    status: false
+  });
+
+  useEffect(async () => {
+    if (adminId !== null) {
```
Comment (DamianPalavecino) [inline src/Components/Admins/form.jsx:17]:
```suggestion
    if (adminId) {
```

File: src/Components/Admins/form.jsx [hunk 10]
```diff
@@ -0,0 +1,177 @@
...
+        },
+        body: JSON.stringify(inputValue)
+      };
+      const url = `${process.env.REACT_APP_API_URL}/admins/${adminId}`;
+      fetch(url, putOptions).then(async (response) => {
+        if (response.status !== 200 && response.status !== 201) {
+          const { message } = await response.json();
+          alert(message);
+        } else {
+          alert('Admin updated successfully');
```
Comment (DamianPalavecino) [inline src/Components/Admins/form.jsx:58]:
```suggestion
          alert(message);
```
Here you can use the message from the backend

File: src/Components/Admins/form.jsx [hunk 11]
```diff
@@ -0,0 +1,177 @@
...
+          status: false
+        })
+      };
+      const url = `${process.env.REACT_APP_API_URL}/admins`;
+      fetch(url, postOptions).then(async (response) => {
+        if (response.status !== 200 && response.status !== 201) {
+          const { message } = await response.json();
+          alert(message);
+        } else {
+          alert('Admin created successfully');
```
Comment (DamianPalavecino) [inline src/Components/Admins/form.jsx:82]: same as above

File: src/Components/Admins/form.jsx (hunk 9 above)
Comment (FabianDutraSilva) [reply src/Components/Admins/form.jsx]: done

File: src/Components/Admins/form.jsx (hunk 11 above)
Comment (FabianDutraSilva) [reply src/Components/Admins/form.jsx]: done

**Review by DamianPalavecino — APPROVED** [review]
Approved, good job

File: src/Components/Admins/form.jsx (hunk 10 above)
Comment (FabianDutraSilva) [reply src/Components/Admins/form.jsx]: excelent correction. Done

**Review by Juanoauil — APPROVED** [review]

**Comment by EliasBermudez13** [comment]
When editing it should show the status that the admin already had (active or inactive), but it's always showing inactive.

**Comment by EliasBermudez13** [comment]
> Al editar, debería mostrar el estado que ya tenía el administrador (activo o inactivo), pero siempre muestra inactivo.
This issue was explained in another ticket to be fixed later after merging. So everything in order, it's approved

**Review by martumoser — APPROVED** [review]

**Pull request merged**